sidecar/register/healthchecker.go

```diff
@@ -71,34 +72,35 @@ func (checker *HealthChecker) Stop() {
 	checker.stop <- struct{}{}
 }
 
-func (checker *HealthChecker) report() {
+// maintainRegistration
+func (checker *HealthChecker) maintainRegistration() {
 	wasHealthy := false // Initialize as unhealthy so that we register on the first successful health check
 
-	statusChan := make(chan HealthStatus, len(checker.checks))
+	statusChan := make(chan healthcheck.Status, len(checker.checks))
 	for _, healthCheck := range checker.checks {
 		healthCheck.Start(statusChan)
 	}
 
 	// Set of health checks that have most recently reported unhealthy statuses.
-	unhealthyStatuses := make(map[HealthCheck]interface{})
+	unhealthyStatuses := make(map[healthcheck.Check]interface{})
```

Comment:
According the Golang docs:

> Interface values are comparable. Two interface values are equal if they have identical dynamic types and equal dynamic values or if both have value nil.

The dynamic value is a pointer (in this case, `*healthcheck.HTTP` or similar). 

For Golang maps:

> The comparison operators == and != must be fully defined for operands of the key type [...]

Which pointers fulfill:

> Pointer values are comparable. Two pointer values are equal if they point to the same variable or if both have value nil. Pointers to distinct zero-size variables may or may not be equal.

You can verify the behavior here:

https://play.golang.org/p/QOTt-EkLi5